plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -1695,6 +1710,27 @@ private void testTimeTravelWithV2Checkpoint(String resourceName)
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    private void testTimeTravelWithV2CheckpointWithTemporalVersion(String resourceName, String v0TimestampWithZone, String v1TimestampWithZone)
```

Comment:
Please consider using a testing resource directory where some of the earliest transaction log files have been removed.

To create the resource files:
```
testing/bin/ptl env up --environment singlenode-delta-lake-oss
```

Further reference
https://docs.delta.io/latest/delta-batch.html#data-retention

Reply:
From the reference, I see the `For example, with a table initially consisting of log entries for versions [0, 19] and a checkpoint at verison 10, if the log entry for version 0 is cleaned up, then you **cannot** time travel to versions [1, 9]`, shall we follow the constraints, currently we are support query [1,9] when 0 is removed since we locate 10 then search towards the head

Reply:
I prefer current workaround, let me know if you have any thoughts.